### PR TITLE
Add symbiflow-examples tutorials for Arty and Basys3 Boards

### PR DESCRIPTION
**Pull request by rw1nkler**
This PR adds the basic `SymbiFlow Examples` tutorial for Arty Board and Basys3 Board

### CODE REVIEW

**Comment by mithro** [comment]
@mgielda / @kgugala / @pgielda -- Do you have any thoughts about how this tutorial should interface with the symbiflow-examples repo? https://github.com/SymbiFlow/symbiflow-examples -- should the symbiflow-examples be a submodule of this one in some way?

**Comment by mithro** [comment]
FYI - @tcal-x

File: source/tutorials/sf-arty.rst [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+   type the command below to add the user to the ``dialout`` group.
+   This should resolve the missing permissions problem:
+
+   .. session:: shell-session
+
+      sudo usermod -a -G dialout `whoami`
+
+   You can also run the ``picocom`` program using ``sudo``
+
+   If the To list all available ttyUSBx devices, use:
```
Comment (tcal-x) [inline source/tutorials/sf-arty.rst:211]: Typo here.

File: source/tutorials/sf-arty.rst [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+3. Check if the design is working correctly.
+
+- You should observe the following line in the OpenOCD output
+
+.. session:: shell-session
+
+   Info : JTAG tap: xc7.tap tap/device found: 0x0362d093 (mfg: 0x049 (Xilinx), part: 0x362d, ver: 0x0)
+
+- In the ``picocom`` terminal, you should observe the following output:
```
Comment (tcal-x) [inline source/tutorials/sf-arty.rst:233]: Maybe it's too early for detailed comments/questions, but is the Linux image being downloaded over Ethernet, or over the USB connector?   I don't really see the instructions for either.   If the image is not loaded, you will instead see the LiteX BIOS prompt when you connect, not a Linux prompt.   But maybe I'm missing something -- I haven't tried actually following this demo step by step.

File: source/tutorials/sf-arty.rst (hunk 1 above)
Comment (rw1nkler) [reply source/tutorials/sf-arty.rst]: Fixed

**Comment by rw1nkler** [comment]
@mithro Can you please, provide the feedback if this type of tutorials is that what you expected?

File: source/tutorials/sf-arty.rst [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        .. session:: shell-session
+
+            pacman -Sy git wget picocom
+
+2. Download and install conda:
+
+.. session:: shell-session
+
+   INSTALL_DIR="/opt/symbiflow/xc7"
+   bash conda_installer.sh -b -p $INSTALL_DIR/conda && rm conda_installer.sh
```
Comment (tcal-x) [inline source/tutorials/sf-arty.rst:62]: Where does "conda_installer.sh" come from?

Comment (rw1nkler) [reply source/tutorials/sf-arty.rst]: Thanks! I've added appropriate information and updated the document

File: source/tutorials/sf-arty.rst (hunk 2 above)
Comment (rw1nkler) [reply source/tutorials/sf-arty.rst]: I've added the appropriate instructions

File: source/tutorials/symbiflow-examples-arty.rst [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+3. Download SymbiFlow Examples
+------------------------------
+
+All the examples used in this tutorial are located in the `SymbiFlow Examples`_
+repository. Download them using:
+
+.. code-block:: bash
+
+   git clone https://github.com/SymbiFlow/symbiflow-examples
```
Comment (tcal-x) [inline source/tutorials/symbiflow-examples-arty.rst:107]: This is too late to clone the repo -- `examples/xc7/environment.yml` is used above, and it lives in the repo.  This needs to come earlier.

File: source/tutorials/symbiflow-examples-arty.rst [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+server, follow the instructions in the :doc:`./howto-install-tftp-server`.
+
+#. Generate the LiteX Linux example bitstream:
+
+   .. code-block:: bash
+
+      cd symbiflow-examples/examples/xc7/linux_litex_demo
+      make
+
+#. Change IP of your Ethernet interface to ``192.168.1.100`` which is expected
```
Comment (tcal-x) [inline source/tutorials/symbiflow-examples-arty.rst:190]: I thought it was 192.168.100.100.

Comment (rw1nkler) [reply source/tutorials/symbiflow-examples-arty.rst]: Fixed

File: source/tutorials/symbiflow-examples-arty.rst (hunk 4 above)
Comment (rw1nkler) [reply source/tutorials/symbiflow-examples-arty.rst]: Fixed

**Comment by aep** [comment]
thanks for this, however following it litteraly doesnt work for me. make in step 6 fails with

/bin/sh: symbiflow_synth: command not found

i have no idea how the artifacts work, but possibly this is not a working build?

    wget -qO- https://storage.googleapis.com/symbiflow-arch-defs/artifacts/prod/foss-fpga-tools/symbiflow-arch-defs/continuous/install/4/20200416-002215/symbiflow-arch-defs-install-a321d9d9.tar.xz | tar -xJ -C $INSTALL_DIR

**Comment by tcal-x** [comment]
> thanks for this, however following it litteraly doesnt work for me. make in step 6 fails with
> 
> /bin/sh: symbiflow_synth: command not found
> 
> i have no idea how the artifacts work, but possibly this is not a working build?
> 
> ```
> wget -qO- https://storage.googleapis.com/symbiflow-arch-defs/artifacts/prod/foss-fpga-tools/symbiflow-arch-defs/continuous/install/4/20200416-002215/symbiflow-arch-defs-install-a321d9d9.tar.xz | tar -xJ -C $INSTALL_DIR
> ```

@rw1nkler , @mithro , the instructions in symbiflow-examples have moved to a newer tar (and newer conda packages), but the instructions here were not updated.   The instructions here seem to duplicate much of the instructions in symbiflow-examples for installing the SymbiFlow toolchain, but they're not automatically updated, so they can get out of sync.   Also, the instructions in symbiflow-examples are tested via tuttest, but these are not (I don't think?) 

@aep , can you include the OS you are working with, and also the specific URL (ideally the direct link to the section) and which command you ran where you ran into problems?   You're working with an Arty A7-35T board, right?

**Comment by aep** [comment]
> @aep , can you include the OS you are working with, and also the specific URL (ideally the direct link to the section) and which command you ran where you ran into problems? You're working with an Arty A7-35T board, right?


the step that fails is https://symbiflow--295.org.readthedocs.build/en/295/tutorials/symbiflow-examples-arty.html#counter-example

yeah, Arty A7-35T. on archlinux.

the instructions in the current master README work fine, they just lack the openocd info

**Comment by tcal-x** [comment]
Thanks @aep!   To confirm, these do work for you:
[https://github.com/SymbiFlow/symbiflow-examples/blob/master/README.rst](https://github.com/SymbiFlow/symbiflow-examples/blob/master/README.rst)
or maybe you found [https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/README.rst](https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/README.rst)

...but the 295 version of the tutorial doesn't work.   Correct?

**Comment by aep** [comment]
yes, i did follow https://github.com/SymbiFlow/symbiflow-examples/blob/master/README.rst and its working fine, thanks :)

but it would be really useful if that contained the info of this PR, specifically the openocd command.

**Comment by rw1nkler** [comment]
Thank you for all your comments! Sorry, for the confusion caused by this stale PR.

We are working on separate sphinx-based documentation for the [symbiflow-examples](https://github.com/SymbiFlow/symbiflow-examples) repository. Soon all the changes from this PR will be incorporated into the new documentation for that repository.

**Pull request closed without merging**